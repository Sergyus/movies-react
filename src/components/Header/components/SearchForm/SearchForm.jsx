import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../../Header.scss';

const SearchForm = ({ fetchSearchMovies, changeFilter }) => {
  const [searchQuery, setSearchQuery] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    if (searchQuery.trim()) {
      changeFilter('Search');
      fetchSearchMovies(1, searchQuery);
    }
  };

  const handleChange = (e) => {
    setSearchQuery(e.target.value);
  };

  return (
    <form className={styles.searchForm} onSubmit={handleSubmit}>
      <input
        aria-label="Search Input"
        placeholder="Search..."
        className={styles.input}
        type="search"
        value={searchQuery}
        onChange={handleChange}
        onBlur={() => setSearchQuery('')}
      />
      <button aria-label="Search Button" type="submit" className={styles.searchBtn} />
    </form>
  );
};

SearchForm.propTypes = {
  fetchSearchMovies: PropTypes.func,
  changeFilter: PropTypes.func,
};

SearchForm.defaultProps = {
  fetchSearchMovies: () => {},
  changeFilter: () => {},
};

export default SearchForm;
