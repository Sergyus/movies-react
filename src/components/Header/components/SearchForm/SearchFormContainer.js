import { connect } from 'react-redux';
import SearchForm from './SearchForm';
import { fetchSearchMovies, changeFilter } from '../../../../modules/app';

const actions = {
  fetchSearchMovies,
  changeFilter,
};

export default connect(null, actions)(SearchForm);
