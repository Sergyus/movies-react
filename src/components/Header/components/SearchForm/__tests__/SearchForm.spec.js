import React from 'react';
import { create, act } from 'react-test-renderer';
import SearchForm from '../SearchForm';

describe('SearchForm component renders correctly', () => {
  const tree = create(<SearchForm />);
  it('SearchForm renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('SearchForm empty onSubmit works correctly', () => {
    const e = { preventDefault: jest.fn() };
    const mockFunction = jest.fn(() => tree.toJSON().props.onSubmit(e));
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });

  it('SearchForm full onSubmit works correctly', () => {
    const input = tree.root.findByProps({ 'aria-label': 'Search Input' });
    const e = { target: { value: '123' } };
    const mockOnChangeFn = jest.fn(() => input.props.onChange(e));
    act(() => {
      mockOnChangeFn();
    });
    const event = { preventDefault: jest.fn() };
    const mockOnSubmitFn = jest.fn(() => tree.toJSON().props.onSubmit(event));
    act(() => {
      mockOnSubmitFn();
    });
    expect(input.props.value).toBe('123');
  });

  it('SearchForm onChange works correctly', () => {
    const input = tree.root.findByProps({ 'aria-label': 'Search Input' });
    const e = { target: { value: '123' } };
    const mockFunction = jest.fn(() => input.props.onChange(e));
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });

  it('SearchForm onBlur works correctly', () => {
    const input = tree.root.findByProps({ 'aria-label': 'Search Input' });
    const mockFunction = jest.fn(input.props.onBlur);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
