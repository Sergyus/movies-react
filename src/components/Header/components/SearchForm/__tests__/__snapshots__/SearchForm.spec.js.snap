// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders correctly SearchForm renders correctly 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    aria-label="Search Input"
    className="input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value=""
  />
  <button
    aria-label="Search Button"
    className="searchBtn"
    type="submit"
  />
</form>
`;
